src/Stratis.Bitcoin.Features.MemoryPool.Tests/TestChainFactory.cs

```diff
@@ -77,16 +78,19 @@ public static async Task<ITestChainContext> CreateAsync(Network network, Script
             network.Consensus.Options = new ConsensusOptions();
             new FullNodeBuilderConsensusExtension.PowConsensusRulesRegistration().RegisterRules(network.Consensus);
 
+            // Dont check PoW of a header in this test.
+            network.Consensus.HeaderValidationRules.RemoveAll(x => x.GetType() == typeof(CheckDifficultyPowRule));
```

Comment:
Interesting approach indeed, I like the creative thinking and the correct usage of rules